independent-projects/extension-maven-plugin/pom.xml

```diff
@@ -34,6 +34,7 @@
         <javax.annotation-api.version>1.3.2</javax.annotation-api.version>
         <javax.inject.version>1</javax.inject.version>
         <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
+        <!-- Not sure exactly why but the Maven plugins need to be compiled with Java 11 -->
         <maven.compiler.target>11</maven.compiler.target>
         <maven.compiler.source>11</maven.compiler.source>
         <maven.compiler.release>11</maven.compiler.release>
```

Comment:
@aloubyansky there's something odd here. If I remove these lines and default to the parent Java 17, I end up with the following error when running the tests (basically anything trying to use the plugin fails):

(I have the same problem with the `quarkus-maven-plugin`, will post the exact error message below)

```
[ERROR]   ExtensionDescriptorMojoTest.shouldExecuteSimplePomCleanly:73->makeMojo:250->AbstractMojoTestCase.lookupConfiguredMojo:461->PlexusTestCase.lookup:205 » ComponentLookup com.google.inject.ProvisionException: Unable to provision, see the following errors:

1) No implementation for BootstrapWorkspaceProvider was bound.
  while locating ExtensionDescriptorMojo
  at ClassRealm[plexus.core, parent: null]
      \_ installed by: WireModule -> PlexusBindingModule
  while locating Mojo annotated with @Named("io.quarkus:quarkus-extension-maven-plugin:999-SNAPSHOT:extension-descriptor")

1 error

======================
Full classname legend:
======================
BootstrapWorkspaceProvider: "io.quarkus.maven.BootstrapWorkspaceProvider"
ExtensionDescriptorMojo:    "io.quarkus.maven.ExtensionDescriptorMojo"
Mojo:                       "org.apache.maven.plugin.Mojo"
Named:                      "com.google.inject.name.Named"
PlexusBindingModule:        "org.eclipse.sisu.plexus.PlexusBindingModule"
WireModule:                 "org.eclipse.sisu.wire.WireModule"
========================
End of classname legend:
========================
```